snippets/window-dependency-injection.md

```diff
@@ -0,0 +1,40 @@
+---
+title: How to inject Window object
+author: fetis26
+level: intermediate
+
+tags:
+  - di
+  - testing
+---
+
+# Content
+For testing purposes you might want to inject Window object in your component.
+You can achieve this with custom `InjectonToken` mechanism provided by Angular.
+
+```typescript
+export const LOCATION_TOKEN = new InjectionToken<Location>('Window location object');
+
+@Component({
+  providers: [
+    { provide: LOCATION_TOKEN, useValue: window.location }
+  ]
+})
+export class SomeComponent {
+  constructor(
+    @Inject(LOCATION_TOKEN) private location: Location
```

Comment:
Wait, why does the title say window, but instead location is injected?
We also need another snippet for window I guess?

Reply:
🤦‍♂️ my fault. the idea is to provide `window.location`